Modules/mathmodule.c

```diff
@@ -2016,6 +2016,71 @@ math_factorial(PyObject *module, PyObject *arg)
 }
 
 
+/*[clinic input]
+math.lcm
+
+    x as a: object
+    y as b: object
+    /
+least common multiple of x and y
+[clinic start generated code]*/
+
+static PyObject *
+math_lcm_impl(PyObject *module, PyObject *a, PyObject *b)
+/*[clinic end generated code: output=6f83fb6d671074ba input=bd41b785dc2a4ff1]*/
+{
+    PyObject *g, *m, *f, *ab;
+    
+    a = PyNumber_Index(a);
+        if (a == NULL) {
+            return NULL;
+    }
+    b = PyNumber_Index(b);
+    if (b == NULL) {
+        return NULL;
+    }
+    if (a == 0 && b == 0) {
+        return 0;
+    }
+    else {
+        if (PyFloat_Check(a) || PyFloat_Check(b)) {
+            PyErr_Format(PyExc_TypeError,"'float' object cannot be an argument to lcm");
+            return 0;  
+        }
+        g = _PyLong_GCD(a, b);
```

Comment:
`_Py_Long_GCD` again returns a new reference.

Owned references before this line: `a`, `b`
Owned references after this line: `a`, `b` and `g` (unless `g == NULL`, in which case just `a` and `b`)